src/main/java/org/wildfly/security/WildFlyElytronProvider.java

```diff
@@ -426,11 +426,11 @@ public Object newInstance(Object constructorParameter) throws NoSuchAlgorithmExc
             if (implementationClass == null) {
                 ClassLoader classLoader = WildFlyElytronProvider.class.getClassLoader();
                 try {
-                    implementationClass = classLoader.loadClass(getClassName());
+                    implementationClass = Class.forName(getClassName(), true, classLoader);
```

Comment:
The second parameter should be set to `false` to avoid excess startup overhead.

Reply:
fixed